rpc/service/src/service.rs

```diff
@@ -493,17 +493,28 @@ NOTE: This error usually indicates an RPC conversion error between the node and
 
         let transaction: Transaction = (&request.transaction).try_into()?;
         let transaction_id = transaction.id();
-        let session = self.consensus_manager.consensus().unguarded_session();
-        let orphan = match allow_orphan {
-            true => Orphan::Allowed,
-            false => Orphan::Forbidden,
-        };
-        self.flow_context.submit_rpc_transaction(&session, transaction, orphan).await.map_err(|err| {
-            let err = RpcError::RejectedTransaction(transaction_id, err.to_string());
-            debug!("{err}");
-            err
-        })?;
-        Ok(SubmitTransactionResponse::new(transaction_id))
+        let inputs = &transaction.inputs;
+
+        let empty_indices = inputs
+            .iter()
+            .enumerate()
+            .filter_map(|(index, input)| (input.signature_script.is_empty()).then_some(index.to_string()))
+            .collect::<Vec<_>>();
```

Comment:
Can be directly joined to the string by
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.join

Reply:
It’s joined into the string when producing an error. I would rather test Vec and if not empty join as opposed to join and then test string. (saves one fn call).

This is how @KaffinPX had it, I asked to move join inside the if.

Reply:
u can call next to understand if it has at least one element and then chain it back. Join method allocates one big string. Current solution allocates many small string, then allocates vector of them, then allocates one big string

Reply:
@biryukovmaxim post an example impl.

Reply:
```rust
 let mut empty_indices =
            inputs.iter().enumerate().filter_map(|(index, input)| (input.signature_script.is_empty()).then_some(index.to_string()));
        let first_empty = empty_indices.next();
        if let Some(first_empty) = first_empty {
            use itertools::Itertools;
            Err(RpcError::EmptySignatureScript(transaction_id, iter::once(first_empty).chain(empty_indices).join(", ")))
        } else { ...   
```

it requires adding itertools to cargo toml:
```
itertools.workspace = true
```